dnf-json

```diff
@@ -98,6 +99,9 @@ class Solver():
 
             for key in desc["gpgkeys"]:
                 if key.startswith("-----BEGIN PGP PUBLIC KEY BLOCK-----"):
+                    # Not using with because it needs to be a valid file for the duration. It
+                    # is inside the temporary persistdir so will be cleaned up on exit.
+                    # pylint: disable=consider-using-with
                     keyfile = tempfile.NamedTemporaryFile(dir=keydir, delete=False)
```

Comment:
As long as `delete=False` you can still use `with...` I think? The file won't be removed on close.

Reply:
Possibly, but I think this is clearer to future me :)